From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Rosenberg <drosen@google.com>
Date: Wed, 8 Jul 2020 02:12:37 -0700
Subject: FROMLIST: ext4: Use generic casefolding support

This switches ext4 over to the generic support provided in
the previous patch.

Since casefolded dentries behave the same in ext4 and f2fs, we decrease
the maintenance burden by unifying them, and any optimizations will
immediately apply to both.

Signed-off-by: Daniel Rosenberg <drosen@google.com>
Reviewed-by: Eric Biggers <ebiggers@google.com>

Link: https://lore.kernel.org/r/20200708091237.3922153-5-drosen@google.com
Bug: 138322712
Bug: 161184936
Test: Boots, /data/media is case insensitive
Change-Id: I3a0705278100590df4c7cdd0dcdf945e9f11feb7
---
 fs/ext4/dir.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/fs/ext4/dir.c b/fs/ext4/dir.c
--- a/fs/ext4/dir.c
+++ b/fs/ext4/dir.c
@@ -668,9 +668,3 @@ const struct file_operations ext4_dir_operations = {
 	.release	= ext4_release_dir,
 };
 
-#ifdef CONFIG_UNICODE
-const struct dentry_operations ext4_dentry_ops = {
-	.d_hash = generic_ci_d_hash,
-	.d_compare = generic_ci_d_compare,
-};
-#endif
